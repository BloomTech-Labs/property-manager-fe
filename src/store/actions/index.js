// IMPORTS/INITIALIZATION =========================|
// ================================================|
import axios from 'axios';
import axiosAuth from '../../helpers/axiosAuth';
// ------------------------------------------------|
// AUTH ACTIONS ===================================|
// ================================================|
export const AUTH_REQUEST_START = 'AUTH_REQUEST_START';
export const AUTH_REQUEST_SUCCESS = 'AUTH_REQUEST_SUCCESS';
export const AUTH_REQUEST_FAIL = 'AUTH_REQUEST_FAIL';
// ------------------------------------------------|
// LOGIN / SIGNUP ---------------------------------|
export const auth = url => (email, password) => async dispatch => {
  dispatch({ type: AUTH_REQUEST_START });

  try {
    const res = await axios.post(url, {
      email,
      password
    });

    localStorage.setItem('token', res.data.token);

    dispatch({ type: AUTH_REQUEST_SUCCESS, payload: { token: res } });
  } catch (err) {
    console.error(err);
    dispatch({ type: AUTH_REQUEST_FAIL, payload: { errorMessage: err } });
  }
};
// ------------------------------------------------|
// PROPERTY ACTIONS ===============================|
// ================================================|
export const ADD_PROPERTY_START = 'ADD_PROPERTY_START';
export const ADD_PROPERTY_SUCCESS = 'ADD_PROPERTY_SUCCESS';
export const ADD_PROPERTY_FAIL = 'ADD_PROPERTY_FAIL';
export const GET_PROPERTY_START = 'GET_PROPERTY_START';
export const GET_PROPERTY_SUCCESS = 'GET_PROPERTY_SUCCESS';
export const GET_PROPERTY_FAIL = 'GET_PROPERTY_FAIL';
// ------------------------------------------------|
// CREATE A PROPERTY ------------------------------|
export const createProperty = url => property => async dispatch => {
  dispatch({ type: ADD_PROPERTY_START });
  console.log(property);
  try {
    const res = await axios.post(url, { ...property });

    dispatch({
      type: ADD_PROPERTY_SUCCESS,
      payload: {
        created: res
      }
    });
  } catch (err) {
    console.error(err);

    dispatch({ type: ADD_PROPERTY_FAIL, payload: { errorMessage: err } });
  }
};
// ------------------------------------------------|
// GET PROPERTIES ---------------------------------|
export const getProperties = url => async dispatch => {
  dispatch({ type: GET_PROPERTY_START });

  try {
    const res = await axiosAuth().get(url);

    dispatch({
      type: GET_PROPERTY_SUCCESS,
      payload: {
        properties: res.data
      }
    });
  } catch (err) {
    console.error(err);

    dispatch({ type: GET_PROPERTY_FAIL, payload: { errMsg: err } });
  }
};
// ------------------------------------------------|
// USER ACTIONS ===================================|
// ================================================|

// define your user actions here

// ------------------------------------------------|
