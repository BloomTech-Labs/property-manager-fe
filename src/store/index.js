import { createStore, applyMiddleware, compose } from 'redux';
import thunk from 'redux-thunk';
import logger from 'redux-logger';
import { getFirebase } from 'react-redux-firebase';
import reducer from './reducers';

const middleware =
  process.ENV === 'production'
    ? [thunk.withExtraArgument(getFirebase)]
    : [thunk.withExtraArgument(getFirebase), logger];

export const makeStore = state =>
  createStore(
    reducer,
    state,
    compose(
      applyMiddleware(...middleware),

      /* eslint-disable */
      window.__REDUX_DEVTOOLS_EXTENSION__
        ? window.__REDUX_DEVTOOLS_EXTENSION__()
        : f => f
    )
  );

export default makeStore();
