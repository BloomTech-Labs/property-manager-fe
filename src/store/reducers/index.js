import { combineReducers } from 'redux';
import authReducer from './authReducer';
import propReducer from './propReducer';
import getUserReducer from './getUserReducer';
import workOrderReducer from './workOrderReducer';

export default combineReducers({
  authReducer,
  propReducer,
  getUserReducer,
  workOrderReducer
});
