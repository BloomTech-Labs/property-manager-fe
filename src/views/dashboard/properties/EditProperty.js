import React, { useEffect } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import PropTypes from 'prop-types';
import { navigate } from '@reach/router';
import PropertyForm from '../../../components/Properties/PropertyForm';
// Redux
import { editProperty, getProperty } from '../../../store/actions';

export default function EditProperty({ id }) {
  // set the url for the api calls and pass in the
  // url param from props
  const url = `https://pt6-propman-staging.herokuapp.com/api/properties/${id}`;

  // pass the url into the edit action
  const updateProperty = editProperty(url);

  // setup dispatch to dispatch the action
  const dispatch = useDispatch();

  // when form is submitted dispatch edit action
  const submitFn = property => {
    dispatch(updateProperty(property)).then(() =>
      navigate('/dashboard/properties')
    );
  };

  // grab the property from redux state
  const property = useSelector(state => state.propReducer.property);

  // grab isGettingProperty bool from state
  const isGettingProperty = useSelector(
    state => state.propReducer.isGettingProperty
  );
  // grab isUpdatingProperty bool from state
  const isSubmitting = useSelector(
    state => state.propReducer.isUpdatingProperty
  );

  // initialValues for the edit form
  const propertyFields = {
    name: property.name,
    street: property.street,
    city: property.city,
    state: property.state,
    zip: property.zip,
    status: property.status
  };

  // get the property by id when component mounts
  useEffect(() => {
    dispatch(getProperty(url));
  }, [dispatch, url]);

  return (
    <div className="properties">
      <PropertyForm
        loading={isGettingProperty}
        isSubmitting={isSubmitting}
        initialValues={propertyFields}
        submit={submitFn}
      />
    </div>
  );
}

EditProperty.defaultProps = {
  id: undefined
};

EditProperty.propTypes = {
  id: PropTypes.string
};
