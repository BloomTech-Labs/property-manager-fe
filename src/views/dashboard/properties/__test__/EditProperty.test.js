import React from 'react';
import * as rtl from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import { Provider } from 'react-redux';
import store from '../../../../store';
import EditProperty from '../EditProperty';

afterEach(rtl.cleanup);

test('snapshot EditProperty', () => {
  const { wrapper } = rtl.render(
    <Provider store={store}>
      <EditProperty />
    </Provider>
  );
  expect(wrapper).toMatchSnapshot();
});

test('Renders', () => {
  const { wrapper } = rtl.render(
    <Provider store={store}>
      <EditProperty />
    </Provider>
  );
  expect(wrapper).not.toBeNull();
});
