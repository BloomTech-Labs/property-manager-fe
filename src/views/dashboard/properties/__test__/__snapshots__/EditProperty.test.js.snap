// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot EditProperty 1`] = `undefined`;
